cpp/src/io/csv/type_conversion.cuh

```diff
@@ -332,8 +200,94 @@ int32_t convertStrtoHash(const char * key, long start_idx, long end_idx, uint32_
     return h1;
 }
 
+__host__ __device__ gdf_date32 parseDateFormat(char *data, long start_idx, long end_idx, bool dayfirst);
+__host__ __device__ gdf_date64 parseDateTimeFormat(char *data, long start_idx, long end_idx, bool dayfirst);
+
+typedef struct parsing_opts_ {
```

Comment:
Does this struct need to be a C struct for CFFI reasons? If not, we should make it a C++ struct. You can then provide in-struct default values for each member.

Reply:
Nope, this is an internal struct so it can be C++. I've changed how the CSV reader uses this struct (populating it from read_csv_args once now) so we don't need the default values as we assign each value.